src/utils/roleAssignment.spec.ts

```diff
@@ -0,0 +1,82 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import request from "../request.js";
+import { sinonUtil } from "./sinonUtil.js";
+import { roleAssignment } from './roleAssignment.js';
+
+describe('utils/roleAssignment', () => {
+  const roleDefinitionId = 'fe930be7-5e62-47db-91af-98c3a49a38b1';
+  const userId = '2056d2f6-3257-4253-8cfc-b73393e414e5';
+  const administrativeUnitId = 'fc33aa61-cf0e-46b6-9506-f633347202ab';
+  const unifieRoleAssignmentWithAdministrativeUnitScopeResponse = {
+    "id": "BH21sHQtUEyvox7IA_Eu_mm3jqnUe4lEhvatluHIWb7-1",
+    "roleDefinitionId": "fe930be7-5e62-47db-91af-98c3a49a38b1",
+    "principalId": "2056d2f6-3257-4253-8cfc-b73393e414e5",
+    "directoryScopeId": "/administrativeUnits/fc33aa61-cf0e-46b6-9506-f633347202ab"
+  };
+  const unifieRoleAssignmentWithTenantScopeResponse = {
+    "id": "BH21sHQtUEyvox7IA_Eu_mm3jqnUe4lEhvatluHIWb7-1",
+    "roleDefinitionId": "fe930be7-5e62-47db-91af-98c3a49a38b1",
+    "principalId": "2056d2f6-3257-4253-8cfc-b73393e414e5",
+    "directoryScopeId": "/"
+  };
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post
+    ]);
+  });
+
+  it('correctly assign a directory (Entra ID) role specified by id with administrative unit scope to a user specified by id', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/roleManagement/directory/roleAssignments') {
+        return unifieRoleAssignmentWithAdministrativeUnitScopeResponse;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const unifiedRoleAssignment = await roleAssignment.createEntraIDRoleAssignmentWithAdministrativeUnitScope(roleDefinitionId, userId, administrativeUnitId);
+    assert.deepStrictEqual(postStub.lastCall.args[0].data, {
+      roleDefinitionId: roleDefinitionId,
+      principalId: userId,
+      directoryScopeId: `/administrativeUnits/${administrativeUnitId}`
+    });
+    assert.deepStrictEqual(unifiedRoleAssignment, {
```

Comment:
Let's aim to have one assert per test. Otherwise, we need to add custom messages so that we can understand why a test failed without having to debug it first